src/ralph/lib/transitions/models.py

```diff
@@ -37,8 +44,28 @@ def run_field_transition(instance, transition, field, data={}, **kwargs):
                 for key, value in data.items()
                 if key.startswith(action.name)
             }
+            for k, v in defaults.items():
+                field = get_field_by_relation_path(instance, k)
+                value = v
+                if field.rel:
+                    value = str(field.rel.to.objects.get(pk=v))
+                history_kwargs[str(field.verbose_name)] = value
+
             defaults.update(kwargs)
-            func(**defaults)
+            result = func(**defaults)
+            action_names.append(str(getattr(func, 'verbose_name', '')))
```

Comment:
if func doesn't have verbose name, (prettified) function name should be taken as default:
```
func.__name__.replace('_', ' ').capitalize()
```